src/nrf5_sdk15_platform/flash/ruuvi_nrf5_sdk15_flash.c

```diff
@@ -545,7 +546,16 @@ void ri_flash_purge (void)
     {
         int page = (FSTORAGE_SECTION_START / erase_unit) + p; // erase unit == virtual page size
         rc = sd_flash_page_erase (page);
-        ri_delay_ms (200);
+
+        if (ri_yield_is_interrupt_context)
```

Comment:
This solves the issue only in this specific case, the proper fix is to check the interrupt context inside ri_delay_millis.
https://github.com/ruuvi/ruuvi.drivers.c/blob/1652e6abaada56d651d2d63424bc7b5e44662dc9/src/nrf5_sdk15_platform/yield/ruuvi_nrf5_sdk15_yield.c#L151

Reply:
Added check inside the `ri_delay_ms`